### PR TITLE
`Integrated code lifecycle`: Show result stats in buildqueue

### PR DESCRIPTION
**Pull request by BBesrour**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

We want to give admins/instructors an overview of the build results of the last 7 days.

### Steps for Testing

Prerequisites:
- 1 Admin
- 1 Programming exercise with a few submissions

1. Log in to Artemis
2. Navigate to Server Administration > Build Queue
3. Check the overview section and whether the numbers are correct (you can use the finished build jobs section in the bottom of the page to check how many build jobs in the last 7 days)
4. Navigation to course management > build queue (top right). 
5. Check the overview section and whether the numbers are correct (you can use the finished build jobs section in the bottom of the page to check how many build jobs in the last 7 days)

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

image

## Summary by CodeRabbit

- **New Features**
  - Introduced build job statistics with interactive time spans (daily, weekly, monthly) in the build queue component.
  - Added collapsible section for build statistics visualization.

- **Enhancements**
  - Improved user interface with new styling for build statistics.
  - Integrated chart data for better visualization of build job statistics.

- **Localization**
  - Added new localization strings for statistics in both German and English.

- **Tests**
  - Added integration tests for retrieving build job statistics.

### CODE REVIEW

File: src/main/webapp/i18n/de/buildQueue.json
Comment (JohannesWt) [inline src/main/webapp/i18n/de/buildQueue.json:32]: I would change the translation to "Daten für den definierten Zeitraum werden abgerufen"

File: src/main/webapp/i18n/en/buildQueue.json
Comment (JohannesWt) [inline src/main/webapp/i18n/en/buildQueue.json:32]: I would also change this translation to "Data for the specified time span is fetched" or similar. But I'm not quite sure what the original translation should intend.

**Review by JohannesWt — CHANGES_REQUESTED** [review]
Code Looks good just small questions about the translations

**Review by JohannesWt — APPROVED** [review]
Code 👍

**Review by JohannesStoehr — DISMISSED** [review]
Code looks good.
Why do you not offer this to instructors as well? E.g. the number of builds that failed might be interesting, since this either shows that students still struggle to write compiling code or that there might be something wrong with an exercise

**Review by az108 — DISMISSED** [review]
Demonstrated and explained. Works as expected and LGTM 👍

**Review by EneaGore — DISMISSED** [review]
Tested on TS3. Works as expected!

**Review by coolchock — APPROVED** [review]
tested on ts3, works as expected 👍

**Review by EneaGore — APPROVED** [review]
re-approve

**Review by az108 — DISMISSED** [review]
Changes LGTM, tested on TS3 again and still works as expected 👍

**Review by pzdr7 — DISMISSED** [review]
Code

**Review by az108 — DISMISSED** [review]
reapprove after merge conflict

**Review by JohannesStoehr — DISMISSED** [review]
Reapprove after merge

**Review by JohannesWt — DISMISSED** [review]
Reapprove - test

**Review by coolchock — DISMISSED** [review]
re-approve after merge conflicts were resolved

**Review by EneaGore — APPROVED** [review]
re-approve after merge

**Review by krusche — APPROVED** [review]

**Pull request merged**
